File: python/7-reverse-integer.py
    class Solution:
    def reverse(self, x: int) -> int:
        min_int = -2 ** 31
        max_int = (2 ** 31) - 1

        res = 0
        while x:
            digit = int(math.fmod(x, 10))
            x = int(x / 10)

            if (res > max_int // 10 or 
                (res == max_int // 10 and digit >= max_int % 10)):
                return 0
            if (res < min_int // 10 or 
                (res == min_int // 10 and digit <= min_int % 10)):
                return 0
            res = (res * 10) + digit
            
        return res

        

File: csharp/7-reverse-integer.cs
public class Solution {
    public int Reverse(int x) {
        int reverse = 0;

        while (x != 0) {
           int digit = x % 10;
           x = x / 10;

           if (reverse > Int32.MaxValue / 10 || 
                (reverse == Int32.MaxValue / 10 && digit > 7)) {
                return 0;   
            }

            if (reverse < Int32.MinValue / 10 || 
                (reverse == Int32.MinValue / 10 && digit < -8)) {
                return 0;   
            } 

            reverse = reverse * 10 + digit;
        }
        return reverse;
    }
}